src/darwin/Framework/CHIP/MTROperationalCredentialsDelegate.mm

```diff
@@ -65,15 +65,8 @@
     // Make copies of the certificates, just in case the API consumer
     // has them as MutableData.
     mRootCert = [NSData dataWithData:rootCert];
-    if (mRootCert == nil) {
```

Comment:
It seems like this can still fail, but agree CHIP_ERROR_NO_MEMORY is the wrong error, should this be invalid data or something?

Reply:
It's not really "invalid data".  It's "we tried to copy the NSData and failed", which really is just OOM, I woudl think.

(And this can probably use `[rootCert copy]` as far as that goes.)

Reply:
Yeah `copy` would work too (though technically someone could pass some sort of custom sub-class that overrides `copy`, so I stuck with the existing `dataWithData:`)

Reply:
The return type of `dataWithData:` is declared as non null, so this is just treating it as such.